app/src/main/java/com/exotikosteam/exotikos/activities/TripListActivity.java

```diff
@@ -59,14 +57,19 @@ private void setupListeners() {
             startActivityForResult(i, NewTripActivity.REQUEST_FLIGHT_SELECTION);
         });
 
-        ItemClickSupport.addTo(rvTrips).setOnItemClickListener(new ItemClickSupport.OnItemClickListener() {
-            @Override
-            public void onItemClicked(RecyclerView recyclerView, int position, View v) {
-                Intent i = new Intent(TripListActivity.this, MainActivity.class);
-                TripStatus trip = getTripForFlight(flights.get(position));
-                i.putExtra(Constants.PARAM_TRIP, Parcels.wrap(trip));
-                startActivity(i);
-            }
+        flightListAdapter.getItemClickSubject().retry().subscribe(flight -> {
+            Intent i = new Intent(TripListActivity.this, MainActivity.class);
+            TripStatus trip = getTripForFlight(flight);
+            i.putExtra(Constants.PARAM_TRIP, Parcels.wrap(trip));
+            startActivity(i);
+        });
+
+        flightListAdapter.getDeleteSubject().retry().subscribe(flight -> {
+            int index = flights.indexOf(flight);
```

Comment:
please use following method to delete flight com.exotikosteam.exotikos.models.trip.TripStatus#deleteFlight 
more details: https://github.com/travelExotikos/Exotikos/pull/47